postgres_exporter/common/queries_nodemx.yml

```
@@ -101,8 +111,19 @@ ccp_nodemx_cpu:
         description: "CPU limit value in milli cores"
 
 ccp_nodemx_cpucfs:
-  query: "SELECT  monitor.cgroup_scalar_bigint('cpu.cfs_period_us') as period_us, 
-                  case when monitor.cgroup_scalar_bigint('cpu.cfs_quota_us') < 0 then 0 else monitor.cgroup_scalar_bigint('cpu.cfs_quota_us') end as quota_us"
+  query: "SELECT
+  CASE
+  WHEN monitor.cgroup_mode() = 'legacy' THEN
+    monitor.cgroup_scalar_bigint('cpu.cfs_period_us')
+  ELSE
+    (SELECT cgroup_array_bigint[2] FROM monitor.cgroup_array_bigint('cpu.max'))
```

🤔 Can this SELECT go away?

```suggestion
    monitor.cgroup_array_bigint('cpu.max')[2]
```

---

It does look funny. The SELECT returns an array named `cgroup_array_bigint`. If you omit the SELECT, you get an error:
```
postgres=# monitor.cgroup_array_bigint('cpu.max');
ERROR:  syntax error at or near "monitor"
LINE 1: monitor.cgroup_array_bigint('cpu.max');
```
If you try to index on the SELECT, you also get an error:
```
postgres=# SELECT monitor.cgroup_array_bigint('cpu.max')[2];
ERROR:  syntax error at or near "["
LINE 1: SELECT monitor.cgroup_array_bigint('cpu.max')[2];
```

---

Looks like it just needs more parentheses.

```sql
CREATE FUNCTION cbandy(TEXT) RETURNS BIGINT[]
LANGUAGE SQL STABLE STRICT
AS $$
  SELECT ARRAY[1000,2000,3000]
$$;

# SELECT cbandy('');
      cbandy
------------------
 {1000,2000,3000}
(1 row)

# SELECT cbandy('')[1];
ERROR:  syntax error at or near "["
LINE 1: SELECT cbandy('')[1];
                         ^
# SELECT (cbandy(''))[1];
 cbandy
--------
   1000
(1 row)

# SELECT (cbandy(''))[2];
 cbandy
--------
   2000
(1 row)
```

---

Nice! I've made the update.